xilem/examples/variable_clock.rs

```diff
@@ -0,0 +1,267 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! This example uses variable fonts in a touch sensitive digital clock.
+
+use std::time::Duration;
+
+use masonry::parley::{
+    fontique::Weight,
+    style::{FontFamily, FontStack},
+};
+use time::{error::IndeterminateOffset, macros::format_description, OffsetDateTime, UtcOffset};
+use winit::error::EventLoopError;
+use xilem::{
+    view::{
+        async_repeat, button, flex, label, prose, sized_box, variable_label, Axis, FlexExt,
+        FlexSpacer,
+    },
+    Color, EventLoop, EventLoopBuilder, WidgetView, Xilem,
+};
+use xilem_core::fork;
+
+/// The state of the application, owned by Xilem and updated by the callbacks below.
+struct Clocks {
+    /// The font [weight](Weight) used for the values.
+    weight: f32,
+    /// The current UTC offset on this machine.
+    local_offset: Result<UtcOffset, IndeterminateOffset>,
+    /// The current time.
+    now_utc: OffsetDateTime,
+}
+
+/// A possible timezone, with an offset from UTC.
+struct TimeZone {
+    /// An approximate region which this offset applies to.
+    region: &'static str,
+    /// The offset from UTC
+    offset: time::UtcOffset,
+}
+
+fn app_logic(data: &mut Clocks) -> impl WidgetView<Clocks> {
+    let view = flex((
+        // HACK: We add a spacer at the top for Android. See https://github.com/rust-windowing/winit/issues/2308
+        FlexSpacer::Fixed(40.),
+        local_time(data),
+        controls(),
+        // TODO: When we get responsive layouts, move this into a two-column view.
+        TIMEZONES.iter().map(|it| it.view(data)).collect::<Vec<_>>(),
+    ));
+    fork(
+        view,
+        async_repeat(
+            |proxy| async move {
+                // TODO: Synchronise with the actual "second" interval. This is expected to show the wrong second
```

Comment:
Yeah I think this is certainly good enough for an example for now, we're not building an atomic clock here^^